component/src/action_handler/actions/delegator_vote.rs

```diff
@@ -0,0 +1,26 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use penumbra_storage::{StateRead, StateWrite};
+use penumbra_transaction::{action::DelegatorVote, Transaction};
+
+use crate::{
+    governance::{check, execute},
+    ActionHandler,
+};
+
+#[async_trait]
+impl ActionHandler for DelegatorVote {
+    async fn check_stateless(&self, context: Arc<Transaction>) -> Result<()> {
+        check::stateless::delegator_vote(self, context)
+    }
+
+    async fn check_stateful<S: StateRead>(&self, state: Arc<S>) -> Result<()> {
+        check::stateful::delegator_vote(&state, self).await
+    }
+
+    async fn execute<S: StateWrite>(&self, state: S) -> Result<()> {
+        execute::delegator_vote(state, self).await
```

Comment:
Is there a particular benefit to having these be individual bare functions, rather than just part of the `ActionHandler` impl?

Reply:
e.g., would we ever want to call `execute::delegator_vote` outside of the context of `DelegatorVote::execute`?